gpu4pyscf/nac/tdrks.py

```diff
@@ -0,0 +1,346 @@
+        log.info("\n")
+        log.info(
+            "******** LR %s gradients for %s ********",
+            self.base.__class__,
+            self.base._scf.__class__,
+        )
+        log.info("cphf_conv_tol = %g", self.cphf_conv_tol)
+        log.info("cphf_max_cycle = %d", self.cphf_max_cycle)
+        log.info("chkfile = %s", self.chkfile)
+        log.info(f"States ID = {self.states}")
+        log.info("\n")
+        return self
+
+    @lib.with_doc(get_nacv.__doc__)
+    def get_nacv(self, x_yI, EI, singlet, atmlst=None, verbose=logger.INFO):
+        return get_nacv(self, x_yI, EI, singlet, atmlst, verbose)
+
+    def kernel(self, xy_I=None, xy_J=None, E_I=None, E_J=None, singlet=None, atmlst=None):
+
+        logger.warn(self, "This module is under development!!")
+
+        if singlet is None:
+            singlet = self.base.singlet
+        if atmlst is None:
+            atmlst = self.atmlst
+        else:
+            self.atmlst = atmlst
+
+        if self.verbose >= logger.WARN:
+            self.check_sanity()
+        if self.verbose >= logger.INFO:
+            self.dump_flags()
+
+        if xy_I is None or xy_J is None:
+            states = sorted(self.states)
+            I, J = states
+            if I < 0 or J < 0:
+                raise ValueError("Excited states ID should be non-negetive integers.")
+            elif I > 0:
+                raise NotImplementedError("Only for ground-excited states nonadiabatic coupling.")
+            elif I == 0:
+                xy_I = self.base.xy[J-1]
+                E_I = self.base.e[J-1]
+                self.de, self.de_scaled, self.de_etf, self.de_etf_scaled \
+                    = self.get_nacv(xy_I, E_I, singlet, atmlst, verbose=self.verbose)
+                self._finalize()
+            else:
+                raise NotImplementedError("Only for ground-excited states nonadiabatic coupling.")
+        return self.de
+    
+    def get_veff(self, mol=None, dm=None, j_factor=1.0, k_factor=1.0, omega=0.0, hermi=0, verbose=None):
+        """
+        Computes the first-order derivatives of the energy contributions from
+        Veff per atom.
+
+        NOTE: This function is incompatible to the one implemented in PySCF CPU version.
+        In the CPU version, get_veff returns the first order derivatives of Veff matrix.
+        """
+        if mol is None:
+            mol = self.mol
+        if dm is None:
+            dm = self.base.make_rdm1()
+        if omega == 0.0:
+            vhfopt = self.base._scf._opt_gpu.get(None, None)
+            return rhf_grad._jk_energy_per_atom(mol, dm, vhfopt, j_factor=j_factor, k_factor=k_factor, verbose=verbose)
+        else:
+            vhfopt = self.base._scf._opt_gpu.get(omega, None)
+            with mol.with_range_coulomb(omega):
+                return rhf_grad._jk_energy_per_atom(
+                    mol, dm, vhfopt, j_factor=j_factor, k_factor=k_factor, verbose=verbose)
+
+
+    def _finalize(self):
```

Comment:
This looks duplicated to the nac.tdrhf code. Consider to reuse the code:
`_finalize = nac.tdrhf.NAC._finalize`